### PR TITLE
#585: migrate to private ips

### PR DESCRIPTION
**Pull request by lynnliu030**
PR for #585. 
* add private_ip() for AWS and GCP
* use private ips for GCP to GCP data transfer
* Also add fix for #603 

Private ips for GCP throughput profiling is in #605

### CODE REVIEW

File: skyplane/replicate/replicator_client.py [hunk 1]
```diff
@@ -256,12 +256,13 @@ def provision_gateway_instance(region: str) -> Server:
...
         # todo add firewall rules for Azure
         public_ips = [self.bound_nodes[n].public_ip() for n in self.topology.gateway_nodes]
+        private_ips = [self.bound_nodes[n].private_ip() for n in self.topology.gateway_nodes if n.region.split(":")[0] == "gcp"]
         authorize_ip_jobs = []
         authorize_ip_jobs.extend(
             [partial(self.aws.add_ips_to_security_group, r.split(":")[1], public_ips) for r in set(aws_regions_to_provision)]
         )
         if gcp_regions_to_provision:
-            authorize_ip_jobs.append(partial(self.gcp.add_ips_to_firewall, public_ips))
+            authorize_ip_jobs.append(partial(self.gcp.add_ips_to_firewall, public_ips + private_ips))
```
Comment (parasj) [inline skyplane/replicate/replicator_client.py:265]: Do we have to specifically authorize private IPs for GCP to GCP transfers?

**Review by parasj — APPROVED** [review]
LGTM!

**Comment by parasj** [comment]
Can you profile a typical GCP to GCP transfer before+after this change?

**Comment by lynnliu030** [comment]
@parasj Here's an example of transferring 9GB of data from gcp to gcp. 

Before (using public ips): transfer runtime = 17.18s, throughput = 4.23 Gbps 
image

After (using private ips): transfer runtime = 10.29s, throughput = 7.06 Gbps 
image

There's around 1.7x throughput improvement and 40% runtime reduction

File: skyplane/replicate/replicator_client.py (hunk 1 above)
Comment (lynnliu030) [reply skyplane/replicate/replicator_client.py:265]: I was testing GCP to GCP transfer without authorizing private IPs but it seems to get stuck

Comment (parasj) [reply skyplane/replicate/replicator_client.py:265]: OK sounds good, we should keep it then

**Comment by parasj** [comment]
Great! glad to see it helps so much

**Review by sarahwooders — APPROVED** [review]
lgtm - feel free to merge!

**Pull request merged**